Wikipedia/Code/WMFSearchViewController.m

```diff
@@ -190,6 +191,8 @@ - (void)viewDidLoad {
     [self updateUIWithResults:nil];
     [self updateRecentSearchesVisibility:NO];
 
+    self.clearedAllRecentSearches = NO;
```

Comment:
Should this be in `viewWillAppear`? Seems like if you clear searches, leave search, re-enter search, perform a search, leave again, it won't save the last search.